stdlib/map.mc

```diff
@@ -6,10 +6,18 @@
 include "option.mc"
 include "seq.mc"
 
+let mapLength : Map k v -> Int =
+  lam m. mapFoldWithKey (lam acc. lam. lam. addi 1 acc) 0 m
+
+let mapLookupOrElse : (Unit -> v) -> k -> Map k v -> v =
+  mapFindOrElse
+
 let mapLookup : k -> Map k v -> Option v =
   lam k. lam m.
-    match mapMem k m with false then None ()
-    else Some (mapFind k m)
+    let res = mapLookupOrElse (lam. None ()) k m in
```

Comment:
Yes, I see what you mean. I can make the signature `mapLookupOrElse: (v1 -> v2) -> (Unit -> v2) -> Map k v1 -> v2` instead, as you suggest. Should I also rename it? Maybe `mapLookupApplyOrElse`?

Reply:
I think renaming is probably a good idea. I don't think I've seen that version, so that's probably as reasonable a name as any. It's a bit meh to have to take the cost of a function application even for cases where the old `mapLookupOrElse` would suffice, but weighing that against more intrinsics I'd probably lean towards fewer intrinsics at the moment.

Reply:
Well, it is easy to keep the old `mapLookupOrElse` as well, and all of this is temporary in any case. So I'll add both!

Reply:
Unless that makes compilation more complicated (maybe that was what you meant)?

Reply:
Not more complicated, just a little bit more work. Then again, this set of intrinsics is meant to be temporary, so they'll be removed down the line anyway, so I guess it doesn't really matter to have both?